### ISSUE TITLE
Tomcat 9 in docker

### ISSUE THREAD

**Issue by barthanssens** [eclipse/rdf4j#2391]
Check if we can move to Tomcat 9 (which uses servlet API 4), this would also make it easy to use the adoptopenjdk image with Eclipse OpenJ9, which in turn is optimized for using (far) less memory than Hotspot.

**Comment by hmottestad** [comment]
Isn't OpenJ9 the extension of the old IBM java version that was 2x slower than hotspot? Wouldn't it make more sense to move to Graal instead? I believe that we can enable the Graal JIT compiler on java 11+ https://openjdk.java.net/projects/graal/. 

And if we want better memory performance we should look at:
 - moving to spring-boot
 - using a better GC (shenandoah og zGC)

**Comment by barthanssens** [comment]
OpenJ9 indeed traces back to IBM's JDK, though I'm not sure if the performance gap still holds in all cases.
Good idea to use Graal JIT. 
By any chance, would you have a recent benchmark comparing HotSpot / OpenJ9 / Graal ?

**Comment by hmottestad** [comment]
Found this one: https://www.phoronix.com/scan.php?page=article&item=graalvm201-openj920-jvm&num=1

Also I think that going for Graal is going to be better for the future. Oracle is using a lot of of resources to develop Graal and I think in the future it will take over for HotSpot. I'm a bit more sceptical about OpenJ9 and how IBM is planning on allocating a competing amount of resources.

It's not a clear cut performance wise. And the newer JDKs suffer from a little known G1GC bug where if you don't set XMS to be equal to XMX then it does some miscalculations and will slow down GC quite a lot. Couldn't find the reference for that.

**Comment by barthanssens** [comment]
Good to know. And indeed, Graal will eventually replace Hotspot, since maintaining the C++ Hotspot codebase isn't exactly ... painless...

**Comment by aschwarte10** [comment]
I agree to close this ticket in favor of https://github.com/eclipse-rdf4j/rdf4j/issues/5801 (for 6.0)

Tomcat 9 is alreafy available in <= 5.3.0
